src/bugsnag.js

```diff
@@ -175,6 +225,181 @@
     }
   }
 
+  // Setup breadcrumbs for click events
+  function trackClicks() {
+    if(!getBreadcrumbSetting("autoBreadcrumbsClicks", true)) {
+      return;
+    }
+
+    window.addEventListener("click", function(event) {
+      self.leaveBreadcrumb({
+        type: "user",
+        name: "UI click",
+        metaData: {
+          targetText: nodeText(event.target),
+          targetSelector: nodeLabel(event.target)
+        }
+      });
+    });
+  }
+
+  // Setup breadcrumbs for console.log, console.warn, console.error
+  function trackConsoleLog(){
+    if(!getBreadcrumbSetting("autoBreadcrumbsConsole")) {
+      return;
+    }
+
+    function trackLog(level, args) {
+      self.leaveBreadcrumb({
+        type: "log",
+        name: "Console output",
+        metaData: {
+          level: level,
+          values: Array.prototype.slice.call(args)
+        }
+      });
+    }
+
+    enhance(console, "log", function() {
+      trackLog("log", arguments);
+    });
+
+    enhance(console, "warn", function() {
+      trackLog("warn", arguments);
+    });
+
+    enhance(console, "error", function() {
+      trackLog("error", arguments);
+    });
+  }
+
+  // Setup breadcrumbs for history navigation events
+  function trackNavigation() {
+    if(!getBreadcrumbSetting("autoBreadcrumbsNavigation")) {
+      return;
+    }
+
+    // check for browser support
+    if (!history || !history.state || !history.pushState || !window.history.pushState.bind) {
```

Comment:
If `window.history` is undefined in older browsers, I believe the first part of this will still throw, unless we are doing `var history = window.history` someplace I missed?